specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json

```diff
@@ -0,0 +1,1234 @@
+          "description": "The exposed ports on the container instance.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContainerPort"
+          }
+        },
+        "environmentVariables": {
+          "description": "The environment variables to set in the container instance.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnvironmentVariable"
+          }
+        },
+        "instanceView": {
+          "description": "The instance view of the container instance. Only valid in response.",
+          "readOnly": true,
+          "type": "object",
+          "properties": {
+            "restartCount": {
+              "readOnly": true,
+              "type": "integer",
+              "description": "The number of times that the container instance has been restarted."
+            },
+            "currentState": {
+              "readOnly": true,
+              "description": "Current container instance state.",
+              "$ref": "#/definitions/ContainerState"
+            },
+            "previousState": {
+              "readOnly": true,
+              "description": "Previous container instance state.",
+              "$ref": "#/definitions/ContainerState"
+            },
+            "events": {
+              "readOnly": true,
+              "description": "The events of the container instance.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Event"
+              }
+            }
+          }
+        },
+        "resources": {
+          "description": "The resource requirements of the container instance.",
+          "$ref": "#/definitions/ResourceRequirements"
+        },
+        "volumeMounts": {
+          "description": "The volume mounts available to the container instance.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VolumeMount"
+          }
+        }
+      }
+    },
+    "ContainerState": {
+      "description": "The container instance state.",
+      "type": "object",
+      "properties": {
+        "state": {
+          "type": "string",
+          "description": "The state of the container instance."
+        },
+        "startTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The date-time when the container instance state started."
+        },
+        "exitCode": {
+          "type": "integer",
+          "description": "The container instance exit codes correspond to those from the `docker run` command."
+        },
+        "finishTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The date-time when the container instance state finished."
+        },
+        "detailStatus": {
+          "description": "The human-readable status of the container instance state.",
+          "type": "string"
+        }
+      }
+    },
+    "Event": {
```

Comment:
>Event [](start = 5, length = 5)

same here - user should not modify the events received from server. Please mark properties as read-only.

Reply:
Done